src/components/Auth/Register/validationScheme.ts

```diff
@@ -0,0 +1,37 @@
+import { RegisterOptions } from 'react-hook-form';
+
+type validationSchemeTypes = {
+  name: RegisterOptions;
+  email: RegisterOptions;
+  password: RegisterOptions;
+};
+
+export const validationScheme: validationSchemeTypes = {
+  name: {
+    required: 'nameRequired',
+    minLength: {
+      value: 2,
+      message: 'nameMinLength',
+    },
+    maxLength: {
+      value: 40,
+      message: 'nameMaxLength',
+    },
+  },
+  email: {
+    required: 'emailRequired',
+    pattern: {
+      value: /^[\w.%+-]+@[\w.-]+\.[\w]{2,6}$/,
+      message: 'emailMismatch',
+    },
+  },
+  password: {
+    validate: (value: string) => {
+      if (!value) return 'passwordRequired';
+      if (!new RegExp('^.{8,}$').test(value)) return 'passwordMinLength';
+      if (!new RegExp('^(?=.*[A-Za-z])').test(value)) return 'passwordLetterMissed';
+      if (!new RegExp('^(?=.*?[0-9])').test(value)) return 'passwordNumberMissed';
+      if (!new RegExp('^(?=.*[#$@!%&*?])').test(value)) return 'passwordSymbolMissed';
+    },
+  },
+};
```

Comment:
Define the regex patterns as constants to improve code readability and maintainability.

```
const EMAIL_PATTERN = /^[\w.%+-]+@[\w.-]+\.[\w]{2,6}$/;
const PASSWORD_MIN_LENGTH_REGEX = /^.{8,}$/;
const PASSWORD_LETTER_REGEX = /^(?=.*[A-Za-z])/;
const PASSWORD_NUMBER_REGEX = /^(?=.*?[0-9])/;
const PASSWORD_SYMBOL_REGEX = /^(?=.*[#$@!%&*?])/;

const validationScheme: validationSchemeTypes = {
  name: {
    required: 'nameRequired',
    minLength: {
      value: 2,
      message: 'nameMinLength',
    },
    maxLength: {
      value: 40,
      message: 'nameMaxLength',
    },
  },
  email: {
    required: 'emailRequired',
    pattern: {
      value: EMAIL_PATTERN,
      message: 'emailMismatch',
    },
  },
  password: {
    validate: (value: string) => {
      if (!value) return 'passwordRequired';
      if (!PASSWORD_MIN_LENGTH_REGEX.test(value)) return 'passwordMinLength';
      if (!PASSWORD_LETTER_REGEX.test(value)) return 'passwordLetterMissed';
      if (!PASSWORD_NUMBER_REGEX.test(value)) return 'passwordNumberMissed';
      return PASSWORD_SYMBOL_REGEX.test(value) ? undefined : 'passwordSymbolMissed';
    },
  },
};
```